test/unit/math/prim/arr/err/is_nonzero_size_test.cpp

```diff
@@ -0,0 +1,41 @@
+#include <stan/math/prim/arr.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <limits>
+#include <vector>
+
+TEST(ErrorHandlingMatrix, IsNonzeroSizeMatrix) {
+  using stan::math::is_nonzero_size;
+
+  std::vector<double> a;
+  a.push_back(3.0);
+  a.push_back(3.0);
+  a.push_back(3.0);
+  a.push_back(3.0);
+
+  EXPECT_TRUE(stan::math::is_nonzero_size(a));
+
+  a.resize(2);
```

Comment:
I think we can remove this test -- we've already got something with size 4. (just these two lines)

Reply:
Can you clarify? Remove lines 16 and 34?

Reply:
Sorry, remove lines 18 and 19. We've tested with a vector of size 4. We don't need to test also with a vector of size 2. It makes sense to test with a vector of size 0.